addons/sale/static/src/js/product_configurator_mixin.js

```diff
@@ -401,37 +383,11 @@ var ProductConfiguratorMixin = {
 
         // parent exclusions (tell which attributes are excluded from parent)
         _.each(combinationData.parent_exclusions, function (ptav) {
-            if (isPtavInCombination(ptav, combination)) {
-                disable = true;
-            }
             // disable the excluded input (even when not already selected)
             // to give a visual feedback before click
             self._disableInput($parent, ptav);
         });
-
-        // archived variants
-        if (isCombinationInList(filteredCombination, combinationData.archived_combinations)) {
-            disable = true;
-        }
-
-        // if not using dynamic attributes, exclude variants that are deleted
-        if (filteredCombination.length && // compatibility 12.0 list view of variants
-            combinationData.has_dynamic_attributes === false &&
-            combinationData.existing_combinations !== undefined &&
-            !isCombinationInList(filteredCombination, combinationData.existing_combinations)
-        ) {
-            disable = true;
-        }
-
-        this.isSelectedVariantAllowed = !disable;
-        $parent.toggleClass('css_not_available', disable);
-        $parent.find("#add_to_cart").toggleClass('disabled', disable);
-        $parent
-            .parents('.modal')
-            .find('.o_sale_product_configurator_add')
-            .toggleClass('disabled', disable);
```

Comment:
So all this code was breaking stuff ? I am scared by the quantity of code you remove in stable versions :/

Reply:
This code was not removed, it was moved into the callback of the RPC of the appropriate widgets (see `_toggleDisable`).

And the rest of the code above in this method was the main issue of the PR: it duplicated the computation in the JS, so now the RPC is taking care of it.